ntbs-service/Services/NotificationService.cs

```diff
@@ -72,50 +70,41 @@ private async Task UpdateUkBorn(PatientDetails patient)
 
         public async Task UpdateTimelineAsync(Notification notification, ClinicalDetails timeline)
         {
-            context.Attach(notification);
-            notification.ClinicalDetails = timeline;
+            context.Entry(notification.ClinicalDetails).CurrentValues.SetValues(timeline);
 
             await context.SaveChangesAsync();
         }
 
         public async Task UpdateEpisodeAsync(Notification notification, Episode episode)
         {
-            context.Attach(notification);
-            notification.Episode = episode;
+            context.Entry(notification.Episode).CurrentValues.SetValues(episode);
 
             await context.SaveChangesAsync();
         }
 
         public async Task UpdateContactTracingAsync(Notification notification, ContactTracing contactTracing) {
-            context.Attach(notification);
-            notification.ContactTracing = contactTracing;
+            context.Entry(notification.ContactTracing).CurrentValues.SetValues(contactTracing);
 
             await context.SaveChangesAsync();
         }
 
         public async Task UpdatePatientTBHistoryAsync(Notification notification, PatientTBHistory tBHistory)
         {
-            context.Attach(notification);
-            notification.PatientTBHistory = tBHistory;
+            context.Entry(notification.PatientTBHistory).CurrentValues.SetValues(tBHistory);
 
             await context.SaveChangesAsync();
         }
 
         public async Task UpdateSocialRiskFactorsAsync(Notification notification, SocialRiskFactors socialRiskFactors)
         {
             UpdateSocialRiskFactorsFlags(socialRiskFactors);
-            var entry = context.Attach(notification);
             context.Entry(notification).Reference(p => p.SocialRiskFactors).TargetEntry.CurrentValues.SetValues(socialRiskFactors);
-            context.Entry(notification).Reference(p => p.SocialRiskFactors).TargetEntry.State = EntityState.Modified;

```

Comment:
The line above already takes care of marking the relevant properties as modified!